scripts/west_commands/runners/intel_adsp.py

```
@@ -0,0 +1,118 @@
+        self.timeout = timeout
+        self.bin_fw = os.path.join(cfg.build_dir, 'zephyr', 'zephyr.ri')
+
+        self.zephyr_base = os.environ.get('ZEPHYR_BASE')
+        self.cavstool = os.path.join(self.zephyr_base, DEFAULT_CAVSTOOL)
+        self.build_dir = cfg.build_dir
+        self.board_dir = cfg.board_dir
+        self.platform = os.path.basename(self.board_dir)
+
+    @classmethod
+    def name(cls):
+        return 'intel_adsp'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        # Optional:
+        parser.add_argument('--remote-host',
+                            help='hostname of the remote targeting ADSP board')
+        parser.add_argument('-t', '--timeout', type=int, default=60,
+                            help='the pty where stdout goes to')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return IntelAdspBinaryRunner(cfg,
+                                    remote_host=args.remote_host,
+                                    timeout=args.timeout,
+                                    )
+
+    def do_run(self, command, **kwargs):
+        self.logger.info('Starting Intel ADSP runner')
+
+        if command == 'flash':
+            if re.search("intel_adsp_cavs", self.platform):
+                self.require(self.cavstool)
+                self.flash(**kwargs)
+            else:
+                self.run_sim(**kwargs)
+        else:
+            self.logger.info('Only support flash cmd now. Do nothing...')
+
+    def flash(self, **kwargs):
+        # Copy the zephyr and cavstool.py to target remote ADSP host
+        download_cmd = (['scp', f'{self.cavstool}', f'{self.bin_fw}', f'{self.remote_host}:'])
+        self.logger.info(f"cavstool({self.cavstool}), fw('{self.bin_fw})")
+
+        try:
+            self.check_call(download_cmd)
+        except subprocess.CalledProcessError:
+            return 1
```

Please do not swallow exceptions like this. Let the exception propagate up; west should handle the CalledProcessError for you if this happens.

---

Got it. I remove the try/except and let it being handled by west. Please kindly let me know if it is correct way to it, thank you!